### DISCUSSION TITLE
Lesson 7:  question About DeployFundMe script

### DISCUSSION THREAD

**Discussion by Perso77** [Cyfrin/foundry-full-course-cu#77] [Q&A] [answered]
this is a script (DeployFundMe) running on anvil chain.
why do I have 2 contract addresses?
![script deploy](https://github.com/Cyfrin/foundry-full-course-f23/assets/116321001/94f0ab44-9909-467a-bc6a-bcca2e006410)
![fundMe](https://github.com/Cyfrin/foundry-full-course-f23/assets/116321001/caccf6c6-04b7-43b1-bd12-3c9c08e4dfde)
![mock](https://github.com/Cyfrin/foundry-full-course-f23/assets/116321001/845e0807-30f7-4672-8ee6-efff84de74cd)

**Comment by PatrickAlphaC** [comment]
Could you rephrase this question? I'm not sure what you're asking?

```
Wonder, how Mock it was deployed at the same time of FundMe, while DeployfundMe not inherit directly from Mock?
```

**Reply by Perso77** [reply]
I hope the question is clear now

**Comment by alymurtazamemon** [comment, marked as answer]
@Perso77 `DeployFundMe` script creates the `HelperConfig` script contract which either gets the price feed address if the network is testnet, in the case of localhost such as Anvil it first create the mock contract and then return the price feed address;
```solidity
MockV3Aggregator mockPriceFeed = new MockV3Aggregator(
    DECIMALS,
    INITIAL_PRICE
);
```
and after that, the `DeployFundMe` script deployed the `FundMe` contract; due to that, you see two contracts deployed on the anvil.
```solidity
FundMe fundMe = new FundMe(priceFeed);
```

**Reply by Perso77** [reply]
Good summary. thanks for the help

**Reply by cromewar** [reply]
Good catch @alymurtazamemon
